services/web/app/src/Features/User/UserRegistrationHandler.js

```diff
@@ -131,10 +131,12 @@ const UserRegistrationHandler = {
                 if (error) {
                   logger.warn({ err: error }, 'failed to send activation email')
                 }
+                callback(null, user, setNewPasswordUrl)
+              },
+              success => {
+                callback(null, user, setNewPasswordUrl)
```

Comment:
`EmailHandler.sendEmail()` is created by calling `util.callbackify()` in the original `async` function:

https://github.com/overleaf/overleaf/blob/e0d5cf4b4268d37819811f063754ad1d1f72e061/services/web/app/src/Features/Email/EmailHandler.js#L8-L9

 From `callbackify` [docs](https://nodejs.org/api/util.html#utilcallbackifyoriginal):

> Takes an async function (or a function that returns a Promise) and returns a function following the error-first callback style, i.e. taking an `(err, value) => ... callback` as the last argument

Which effectively means any callback except the last one will be ignored: in this case, the first callback (the one calling `logger.warn()` when sending the email fails) is ignored. A quick example:

```javascript
const util = require('util');

const fnError = () => Promise.reject()

const callbackFunction = util.callbackify(fnError)

callbackFunction(
    (err) => {
        if (err) console.error("1st callback");
    },
    (err) => { 
        if (err) console.error("2nd callback");
    },
);

// Outputs "2nd callback" only
```

Removing the 2nd callback should fix the problem: the error will still be logged and the callback is invoked regardless of whether `sendEmail` fails or not:

```suggestion
```

Reply:
Indeed, i removed the 3 lines. Thanks for reviewing.